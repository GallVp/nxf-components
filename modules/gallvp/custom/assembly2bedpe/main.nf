process CUSTOM_ASSEMBLY2BEDPE {
    tag "${meta.id}"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/biopython:1.75':
        'biocontainers/biopython:1.75' }"

    input:
    tuple val(meta), path(assembly)

    output:
    tuple val(meta), path("*.bedpe"), emit: bedpe
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    template('assembly2bedpe.py')

    stub:
    args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    echo ${args}

    touch ${prefix}.bedpe

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version | cut -d' ' -f2)
        biopython: \$(pip list | grep "biopython" | cut -d' ' -f3)
    END_VERSIONS
    """
}
