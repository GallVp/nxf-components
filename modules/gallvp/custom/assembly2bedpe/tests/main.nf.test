nextflow_process {

    name "Test Process CUSTOM_ASSEMBLY2BEDPE"
    script "../main.nf"
    config "./nextflow.config"
    process "CUSTOM_ASSEMBLY2BEDPE"

    tag "modules"
    tag "modules_gallvp"
    tag "custom"
    tag "custom/assembly2bedpe"

    test("inline - assembly") {

        when {
            process {
                """
                def test_file = new File('inline.assembly')
                test_file.text = ">HYh1_Chr01:6000000-6015000 1 15001\\n>HYh2_Chr01:6000000-6015000 2 15001\\n1\\n2"

                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    test_file.toPath()
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("inline - assembly - stub") {

        options '-stub'

        when {
            process {
                """
                def test_file = new File('inline.assembly')
                test_file.text = ">HYh1_Chr01:6000000-6015000 1 15001\\n>HYh2_Chr01:6000000-6015000 2 15001\\n1\\n2"

                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    test_file.toPath()
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("inline - assembly - single scaffold") {

        when {
            params {
                assembly2bedpe = '-s'
            }
            process {
                """
                def test_file = new File('inline.assembly')
                test_file.text = ">HYh1_Chr01:6000000-6015000 1 15001\\n>HYh2_Chr01:6000000-6015000 2 15001\\n1\\n2"

                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    test_file.toPath()
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
