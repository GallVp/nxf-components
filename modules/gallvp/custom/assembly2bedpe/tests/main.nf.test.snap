{
    "inline - assembly": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bedpe:md5,ae7a65d0efada907de477d43afcf2f0d"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bed:md5,192aeedad96d028fc0f1e88099c95525"
                    ]
                ],
                "2": [
                    "versions.yml:md5,704c8164e659d86df96218632b156d7a"
                ],
                "bed": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bed:md5,192aeedad96d028fc0f1e88099c95525"
                    ]
                ],
                "bedpe": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bedpe:md5,ae7a65d0efada907de477d43afcf2f0d"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,704c8164e659d86df96218632b156d7a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-07-21T16:04:57.123834"
    },
    "inline - assembly - single scaffold": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bedpe:md5,511c63569e96a71cc3c14d13c7198f15"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bed:md5,ff0dc82bcd1f6e19089f1ab8a1a35a16"
                    ]
                ],
                "2": [
                    "versions.yml:md5,704c8164e659d86df96218632b156d7a"
                ],
                "bed": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bed:md5,ff0dc82bcd1f6e19089f1ab8a1a35a16"
                    ]
                ],
                "bedpe": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bedpe:md5,511c63569e96a71cc3c14d13c7198f15"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,704c8164e659d86df96218632b156d7a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-07-21T16:05:04.671481"
    },
    "inline - assembly - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bedpe:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,704c8164e659d86df96218632b156d7a"
                ],
                "bed": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "bedpe": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bedpe:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,704c8164e659d86df96218632b156d7a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-07-21T16:04:24.260298"
    }
}